AutomatedTesting/Gem/PythonTests/scripting/NodeCategory_ExpandOnClick.py

```diff
@@ -45,79 +52,83 @@ def NodeCategory_ExpandOnClick():
-
-    # 2) Get the SC window objects
-    editor_window = pyside_utils.get_editor_main_window()
-    sc = editor_window.findChild(QtWidgets.QDockWidget, "Script Canvas")
-    if sc.findChild(QtWidgets.QDockWidget, "NodePalette") is None:
-        action = pyside_utils.find_child_by_pattern(sc, {"text": "Node Palette", "type": QtWidgets.QAction})
-        action.trigger()
-    node_palette = sc.findChild(QtWidgets.QDockWidget, "NodePalette")
-    nodeTree = node_palette.findChild(QtWidgets.QTreeView, "treeView")
-    ai_index = pyside_utils.find_child_by_pattern(nodeTree, "AI")
-
-    # 3) Ensure all categories are collapsed for a clean state
-    nodeTree.collapseAll()
-
-    # 4) Left-Click on a node category arrow to expand it
-    left_click_arrow(nodeTree, ai_index)
-
-    # 5) Verify it expanded
-    Report.result(Tests.click_expand, nodeTree.isExpanded(ai_index))
-
-    # 6) Left-Click on a node category arrow to collapse it
-    left_click_arrow(nodeTree, ai_index)
-
-    # 7) Verify it collapsed
-    Report.result(Tests.click_collapse, not nodeTree.isExpanded(ai_index))
-
-    # 8) Double-Click on a node category to expand it
-    double_click(nodeTree, ai_index)
-
-    # 9) Verify it expanded
-    Report.result(Tests.dClick_expand, nodeTree.isExpanded(ai_index))
-
-    # 10) Double-Click on a node category to collapse it
-    double_click(nodeTree, ai_index)
-
-    # 11) Verify it collapsed
-    Report.result(Tests.dClick_collapse, not nodeTree.isExpanded(ai_index))
-
-
-if __name__ == "__main__":
-    import ImportPathHelper as imports
-
-    imports.init()
-    from utils import Report
+        pyside_utils.item_view_index_mouse_click(node_palette_node_tree, index)
+        QtTest.QTest.mouseDClick(node_palette_node_tree.viewport(),
+                                 QtCore.Qt.LeftButton,
+                                 QtCore.Qt.NoModifier,
+                                 item_index_center)
+
+    @pyside_utils.wrap_async
+    async def run_test(self):
+
+        # Preconditions
+        general.idle_enable(True)
+        general.close_pane(SCRIPT_CANVAS_UI)
+        helper.wait_for_condition(lambda: general.is_pane_visible(SCRIPT_CANVAS_UI) is None, WAIT_TIME_3)
+
+        # 1) Open Script Canvas pane
+        general.open_pane(SCRIPT_CANVAS_UI)
+        Report.critical_result(Tests.pane_open, general.is_pane_visible(SCRIPT_CANVAS_UI))
+
+        # 2) Get the SC window objects (editor, sc editor, node palette elements)
+        scripting_tools.initialize_editor_object(self)
+        scripting_tools.initialize_sc_editor_objects(self)
+        scripting_tools.open_node_palette(self)
+        #wait for the node palette and other SC elements to render
+        helper.wait_for_condition(lambda: self.sc_editor.findChild(QtWidgets.QDockWidget, NODE_PALETTE_QT) is not None, WAIT_TIME_3)
+        node_palette_node_tree = scripting_tools.get_node_palette_node_tree_qt_object(self)
```

Comment:
White space before comment and after the wait for condition statement

Reply:
reformatted to make it clearer